pkg/eniconfig/eniconfig.go

```diff
@@ -227,3 +114,32 @@ func getEniConfigLabelDef() string {
 	}
 	return defaultEniConfigLabelDef
 }
+
+func GetNodeSpecificENIConfigName(ctx context.Context, k8sClient client.Client) (string, error) {
+    var eniConfigName string
+	nodeList := corev1.NodeList{}
+	err := k8sClient.List(ctx, &nodeList)
+	if err != nil {
+		fmt.Errorf("Error while Node List Get: %s", err)
+	}
+	log.Debugf("Node Count: ", len(nodeList.Items))
+	for _, node := range nodeList.Items {
+		if node.Name == os.Getenv("MY_NODE_NAME") {
```

Comment:
Why is this a List() - can we just call Get(MY_NODE_NAME) ?

Reply:
Yeah, repurposed legacy code. Will change it.